### PR TITLE
Add a recipe to build scikit-image

### PR DESCRIPTION
**Pull request by jakirkham**
This adds a recipe to build scikit-image. It is loosely based off of the one from conda-recipes, but has been modified quite a bit.

Note: `conda skeleton pypi` did not work here and the recipe in conda-recipes is pretty out of date. So, this may need a bit of love.

### CODE REVIEW

**Comment by ivoflipse** [comment]
You can add me as a maintainer for this one and Korijn as well. That way we don't have to maintain our own versions

**Comment by msarahan** [comment]
Please add me as a maintainer.  Win fail looks like a transient download failure with anaconda.org.  Mac fail looks like rpath link options failed.  Do we need those, or does conda build take care of it post-build?

**Comment by jakirkham** [comment]
> Mac fail looks like rpath link options failed. Do we need those, or does conda build take care of it post-build?

Was having an issue with fixing rpaths as we were running out of space somehow. This was an attempt to fix that, but it obviously needs some tweaks.

File: recipes/scikit-image/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  version: {{ version }}
+
+source:
+  fn: scikit-image-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/s/scikit-image/scikit-image-{{ version }}.tar.gz
+  md5: 04ea833383e0b6ad5f65da21292c25e1
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt  # [not osx]
```
Comment (ocefpaf) [inline recipes/scikit-image/meta.yaml]: I don't have a strong feeling about this, but it feels weird to have the build commands split in two files. I would either eliminate this line as it is not needed. Or I would removed the `build.sh` and add a `export LDFLAGS="-L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"  # [osx]` line here.

Comment (jakirkham) [inline recipes/scikit-image/meta.yaml]: Ok, I'll eliminate it. Thanks for the feedback.

File: recipes/scikit-image/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [ `uname` == "Darwin" ]
+then
+    export LDFLAGS="-L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"
```
Comment (jakirkham) [inline recipes/scikit-image/build.sh]: Turns out Mac doesn't like `-rpath=`, but prefers `-rpath,` as [mentioned on SO]( http://stackoverflow.com/a/32400674 ). Seems to be working now that it is fixed.

File: recipes/scikit-image/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  home: http://scikit-image.org/
+  license: BSD 3-Clause
+  summary: Image processing routines for SciPy
+
+extra:
+  recipe-maintainers:
+    - ivoflipse
+    - jakirkham
+    - Korijn
+    - msarahan
```
Comment (jakirkham) [inline recipes/scikit-image/meta.yaml]: Added @ivoflipse and @msarahan per request. Was suggested by @ivoflipse to add @Korijn too. Is this ok @Korijn?

**Comment by jakirkham** [comment]
> Mac fail looks like rpath link options failed. Do we need those, or does conda build take care of it post-build?

@msarahan, as you can see in this log [line]( https://travis-ci.org/conda-forge/staged-recipes/jobs/131972607#L7390 ) we get a failure trying to fix the rpath with `install_name_tool`. I suppose we could use `-headerpad_max_install_names`. Honestly, can't remember whether I tried that or not. I think I tried `detect_binary_files_with_prefix` which didn't work so thought that approach was out. This strategy seems pretty clean to me as it just sets the rpath for us as part of the compilation process and it appears to be working.

File: recipes/scikit-image/build.sh (hunk 2 above)
Comment (jakirkham) [inline recipes/scikit-image/build.sh]: Appears this doesn't cut it actually. Try building with Python 3.5 and NumPy 1.10.x (though presumably the problem would occur with NumPy 1.11.x). Here is the [log]( https://api.travis-ci.org/jobs/131975942/log.txt?deansi=true ) for details. Though `-headerpad_max_install_names` seems to still work fine.

**Comment by jakirkham** [comment]
Alright, this is passing and ready to go. 😄

**Comment by ocefpaf** [comment]
Let the feedstock fly!

File: recipes/scikit-image/meta.yaml (hunk 3 above)
Comment (Korijn) [inline recipes/scikit-image/meta.yaml]: Yes, perfect.

**Comment by jakirkham** [comment]
Thanks!

**Comment by jakirkham** [comment]
@ahojnnes @stefanv @blink1073 @jni @vighneshbirodkar @cgohlke, just letting you know that we have added `scikit-image` to [conda-forge]( http://conda-forge.github.io/ ). This will allow users to easily install this with [`conda`]( http://conda.pydata.org/docs/ ) ( [`conda install -c conda-forge scikit-image`]( http://anaconda.org/conda-forge/scikit-image ) ). The maintenance burden here is light as all releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe in [this feedstock (repo)]( https://github.com/conda-forge/scikit-image-feedstock ). If you have any interest in being added as a maintainer (to make new releases for instance), we would be more than happy to add any of you that are interested to the maintainers list.

**Comment by jni** [comment]
Thanks for this!

I'd be very interested, but I definitely need a crash course in conda recipe development. I have not yet found a good, comprehensive source for all that one needs to know: what are all the possible fields and their possible values, how does one specify cross-platform builds, etc. Do you guys have an "internal" documentation that you have developed for this?

Thanks!

**Comment by stefanv** [comment]
Thanks for doing this, everyone!
​

**Comment by ahojnnes** [comment]
This looks great, thanks from my side as well!

**Comment by Korijn** [comment]
@jni there is a lot of public documentation! The most important article for recipe development being the following:

http://conda.pydata.org/docs/building/meta-yaml.html

Other than that, it's still kind of trial & error since it's under development. I found it helpful to "watch" the conda, conda-env and conda-build repos on github.

File: recipes/scikit-image/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  number: 0
+
+requirements:
+  build:
+    - setuptools
+    - cython
+    - python
+    - six
+    - numpy x.x
```
Comment (Korijn) [inline recipes/scikit-image/meta.yaml]: @jakirkham care to explain what the `x.x` bit means?

Comment (jakirkham) [inline recipes/scikit-image/meta.yaml]: Yeah. So, this is a particular oddity in `conda-build`. That specifies we are linking to `numpy` and we expect to match the same major and minor version from build-time at run-time. There has been discussion about expanding this to other packages, as well.

**Comment by jakirkham** [comment]
Glad to have such a positive response here. 😄 Sorry I'm so slow to respond. 😞

> I have not yet found a good, comprehensive source for all that one needs to know: what are all the possible fields and their possible values, how does one specify cross-platform builds, etc.

I second @Korijn that the [`meta.yaml`]( http://conda.pydata.org/docs/building/meta-yaml.html ) documentation is a great place to start, @jni. This definitely covers all of the fields available and how they can be set.

Beyond that I would recommend looking at our [wiki]( https://github.com/conda-forge/staged-recipes/wiki ) for some more tips about recipes in particular. We are in the early stages of working on some more [general docs]( http://conda-forge.readthedocs.io/ ) to describe how things work here. Currently this includes some guidelines about additions. Though we could include relevant bits about conda-recipes.

Though one of the best things to do (as with most things) is see some examples. There are loads of recipes here, which would be great for learning from. So, feel free to pull relevant pieces from other recipes when trying to make new ones.

Beyond this, if you still have questions, please ask. We are happy to help.

**Pull request merged**